packages/web/src/pages/App.js

```diff
@@ -466,52 +439,36 @@ class App extends Component {
         />
       )
 
-    const showBanner = !!(
-      showCTABanner ||
-      showWeb3ErrorBanner ||
-      showWebUpdateBanner
-    )
-
     const SwitchComponent = isMobile() ? AnimatedSwitch : Switch
     const noScroll = matchPath(this.state.currentRoute, CHAT_PAGE)
 
     return (
-      <DesktopHeaderContext.Provider value={{ offsetForBanner: showBanner }}>
-        {this.props.isChatEnabled ? <ChatListener /> : null}
-        <div className={cn(styles.app, { [styles.mobileApp]: isMobileClient })}>
-          {showCTABanner ? (
-            <MobileDesktopBanner
-              onClose={this.dismissCTABanner}
-              onAccept={this.showDownloadAppModal}
-            />
-          ) : null}
-          {showWeb3ErrorBanner ? (
-            <Web3ErrorBanner
-              alert
-              isElectron={client === Client.ELECTRON}
-              onClose={this.dismissWeb3ErrorBanner}
-            />
-          ) : null}
+      <div className={styles.root}>
+        <AppBannerWrapper>
+          <DownloadAppBanner />
+          <DirectMessagesBanner />
+          <Web3ErrorBanner />
+          {/* Other banners' logic is self-contained, but since this one uses the IPC 
+            and can result in either required or optional updates, keeping the visibility 
+            controlled from this parent for now */}
           {showWebUpdateBanner ? (
             <UpdateAppBanner
               onAccept={this.acceptWebUpdate}
               onClose={this.dismissUpdateWebAppBanner}
             />
           ) : null}
+        </AppBannerWrapper>
+        {this.props.isChatEnabled ? <ChatListener /> : null}
```

Comment:
Should the DMs banner also care about isChatEnabled? Or maybe we just lose this here?

Reply:
Very good catch - I don't think we need a DMs kill switch, maybe a follow up pr to strip out the checks for `isChatEnabled` is in order